### ISSUE TITLE
What happened to binary releases?

### ISSUE THREAD

**Issue by cmichi** [WebAssembly/binaryen#4148]
In the past you had always attached the binary build for various platforms to a release. What's the reason for stopping in the latest release?

This impacts us, since we used these binary releases in our CI setup. I only noticed that they no longer exist since our CI broke.

Moreover, we often recommended installing the binary release directly in our community. A number of `stable` Linux distributions still contain ancient `binaryen` releases, so this was an easy way to get the latest release for e.g. Ubuntu LTS (IIRC).

Yeah, so would be great if you could add them back :-).

**Comment by kripken** [comment]
I don't think we changed anything intentionally, but yes, the new version 102 lacks binaries.

Looks like builds failed:

https://github.com/WebAssembly/binaryen/actions/runs/1222083042

I'm not very familiar with github actions so those errors don't immediately suggest anything to me...

@sbc100 any ideas?

**Comment by sbc100** [comment]
Looks like the "upload-release-asset" step is failing for each of the builders.  

https://github.com/WebAssembly/binaryen/blob/5b90e0332253ee879d16fbc29d391ad75734ecf5/.github/workflows/build_release.yml#L69-L77

I have no clues yet as to why.

**Comment by maurolacy** [comment]
Looks like a temporary failure to me. I would just retry it, at least once.

**Comment by cmichi** [comment]
Just noticed that another release was created today and the GHA failed for that one as well: https://github.com/WebAssembly/binaryen/actions/runs/1233610300.

**Comment by sbc100** [comment]
Indeed, looks like the upload recipe we were using has become unmaintained: https://github.com/WebAssembly/binaryen/pull/4151

**Comment by sbc100** [comment]
Well I tried 3 different alternative github action and they all failed in different ways :(

For now I guess we could download the aritfacts form the CI builder and then upload them manually to the release.. :(

**Comment by sbc100** [comment]
You can find the CI artifacts here: https://github.com/WebAssembly/binaryen/actions/runs/1222068571

Do those work for you?

**Comment by maurolacy** [comment]
Looks like a timeout error or so.

Lots of info / workarounds [here](https://github.com/request/request/issues/2047) and [here](https://github.com/request/request/issues/2047).

**Comment by maurolacy** [comment]
> You can find the CI artifacts here: https://github.com/WebAssembly/binaryen/actions/runs/1222068571
> 
> Do those work for you?

Not for us, as we are interested in the alpine-based build. I guess we'll switch to compiling our own version for both archs (Amd64 and Arm64).

**Comment by cmichi** [comment]
Is there any update on this?

@sbc100 https://github.com/softprops/action-gh-release seems to be well maintained and supports uploading assets, so maybe it's possible to switch to that GHA?

**Comment by sbc100** [comment]
I'm giving it a try in #4362 but so far its not looking good.  Seeing #403 errors.

**Comment by sbc100** [comment]
Looks like it works!

**Comment by cmichi** [comment]
> Looks like it works!

Hey that's great! Would this be a good opportunity for another release? 🙃

**Comment by tlively** [comment]
Looks like version 111 is also missing binary assets. It doesn't look like a workflow was even kicked off for it: https://github.com/WebAssembly/binaryen/actions/workflows/create_release.yml

**Comment by sbc100** [comment]
I think maybe you have to push the tag.. not create it directly in the UI.  Can you try deleting the tag and pushing it from your local checkout?

**Comment by tlively** [comment]
I did create the tag locally and push it up. I only used the UI to create the release from that tag.

**Comment by tlively** [comment]
I recreated the `version_111` release using the GitHub UI rather than from an existing tag and that successfully kicked off the builders.
